docs/architecture/adr-069-flexible-node-intitalization.md

```diff
@@ -0,0 +1,164 @@
+# ADR 069: Flexible Node Initialization
+
+## Changlog
+
+- 2021-06-09: Initial Draft (@tychoish)
+
+## Status 
+
+Proposed. 
+
+## Context
+
+In an effort to support [Go-API-Stability](./adr-060-go-api-stability.Md),
+during the 0.35 development cycle, we have attempted to reduce the total size
+of the API surface area by moving most of the interface of the `node` package
+into unexported functions, as well as moving the reactors to an `internal`
+package. Having this coincide with the 0.35 release made a lot of sense
+because these interfaces were _already_ changing as a result of the `p2p`
+[refactor](./adr-061-p2p-refactor-scope.md), so it made sense to think a bit
+more about how tendermint exposes this API.
+
+It is not immediately intuitive that the nexus of this change is in the
+relationship between the `rpc` package and the `node` package, particularly
+with regards to how the node instances are started. However, this is indeed
+the case and the existing dependencies are tangled around supporting these
+cases.
+
+## Observations
+
+To prepare for this work [this
+branch](https://github.com/tendermint/tendermint/tree/tychoish/scratch-node-minimize)
+contains some experimentation with the node implementation to see if,
+without making major modifications or changing interfaces, what
+aspects could be simplified. To summarize, generally:
+
+- The implementation of the `node` service, doesn't really need the
+  ability to introspect the implementations of reactors in most
+  cases.
+
+- The exception to this, is the cluster of
+  consensus/statesync/blockhain (which depend on eachother,) because
+  `node` itself contains some of their specific initialization logic.
+
+- While the RPC environment construction happens in `node`, this can
+  happen early during `node` construction rather than at
+  initialization time, which reduces the need for the implementation
+  of the node object to track this state.
+
+Ideally, it seems like a node instance should be expressable simply as
+a list of `service.Services` and any knowledge of the internals of any
+of these services could be limited to their constructors. There is a
+sketch of a generic and simple `groupService` implementation, to handle a
```

Comment:
I agree with the goal here ("it seems like a node instance should be expressable simply as a list of `service.Services` and any knowledge of the internals of any of these services could be limited to their constructors") but I don't understand how the `groupService` can be used to support this. I checked the `scratch-node-minimize` branch and didn't see it in use

Reply:
I hadn't quite gotten there in my branch to wire things together. I think the `groupService` thing can fall out except in the detailed design.